lib/elixir/lib/module.ex

```diff
@@ -345,8 +345,9 @@ defmodule Module do
   ### Custom attributes
 
   In addition to the built-in attributes outlined above, custom attributes may
-  also be added. A custom attribute is any valid identifier prefixed with an
-  `@` and followed by a valid Elixir value:
+  also be added. Custom attributes are expressed using the `@` operator followed
+  by a variable. The value given to the custom attribute must be a valid Elixir 
```

Comment:
Maybe better?

```suggestion
  by a variable expression. The value given to the custom attribute must be a valid Elixir 
```

Reply:
name perhaps?

```suggestion
  by a variable name. The value given to the custom attribute must be a valid Elixir 
```

Reply:
I used "expression" because that's the term we use in the "Syntax reference" guide. https://hexdocs.pm/elixir/syntax-reference.html#expressions

Wdyt?

> In addition to the built-in attributes outlined above, custom attributes may also be added. 
>
> Custom attributes are expressed using the `@` operator followed by a variable expression. 
> See the [Syntax Reference](syntax-reference.html) documentation for more information.
>
> The value given to the custom attribute must be a valid Elixir value:

Reply:
The module attribute syntax allows @ followed by **any expression**, not only variable ones. But the default implementation allows only variable names. So we need to fix the syntax reference too. :) I would simply say:

> Custom attributes are expressed using the @ operator followed by an expression.

Reply:
I went with "valid variable name". "an expression" might be more accurate, but I think "a valid variable name" is going to be less confusing for final users 🙂